request/src/main/java/com/vimeo/networking2/VimeoService.kt

```diff
@@ -228,249 +220,276 @@ interface VimeoService {
-        @QueryMap options: Map<String, String>,
-        @Body body: Any
-    ): VimeoCall<User>
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>,
+        @Header(CACHE_CONTROL) cacheControl: CacheControl?
+    ): VimeoCall<Unit>
 
     @PUT
     fun putContentWithUserResponse(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @QueryMap options: Map<String, String>
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>,
+        @Body body: Any?
     ): VimeoCall<User>
 
-    @GET
-    fun getUnit(
+    @PUT
+    fun putContentWithUserResponse(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @QueryMap options: Map<String, String>,
-        @Header(CACHE_CONTROL) cacheControl: CacheControl?
-    ): VimeoCall<Unit>
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>
+    ): VimeoCall<User>
 
     @PUT
     fun put(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @QueryMap options: Map<String, String>
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>
     ): VimeoCall<Unit>
 
     @PUT
     fun put(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @Header(CACHE_CONTROL) cacheControl: CacheControl?,
-        @QueryMap options: Map<String, String>,
-        @Body body: Any
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>,
+        @Body body: Any?
     ): VimeoCall<Unit>
 
     @DELETE
     fun delete(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @QueryMap options: Map<String, String>
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>
     ): VimeoCall<Unit>
 
     @POST
     fun post(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @Header(CACHE_CONTROL) cacheControl: CacheControl?,
-        @Body parameters: List<Any>
+        @Body parameters: List<@JvmSuppressWildcards Any>
     ): VimeoCall<Unit>
 
-    @Headers(HEADER_NO_CACHE)
     @PATCH
     fun emptyResponsePatch(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @QueryMap options: Map<String, String>,
+        @QueryMap options: Map<String, @JvmSuppressWildcards String>,
         @Body parameters: Any
     ): VimeoCall<Unit>
 
     @POST
     fun emptyResponsePost(
         @Header(AUTHORIZATION) authorization: String,
         @Url uri: String,
-        @Body parameters: Map<String, String>
+        @Body parameters: Map<String, @JvmSuppressWildcards String>
     ): VimeoCall<Unit>
 
     @Headers(HEADER_NO_CACHE)
     @GET("search")
     fun search(
         @Header(AUTHORIZATION) authorization: String,
-        @QueryMap queryParams: Map<String, String>
+        @QueryMap queryParams: Map<String, @JvmSuppressWildcards String>
```

Comment:
There are two generic parameter types supported in this service `@QueryMap` and `@Body`. Ideally, we wouldn't have any generic parameters and we would have the fully supported type here. However, we don't have the time to do that for this iteration, so we have to live with the generic parameters for now. The issue I have run into is regarding naming and the supported types.

`@QueryMap` is always of the type `Map<String, String>`, but the name differs in this file between `queryParams` and `options`. Additionally, it is almost always called `refinementMap` in the `VimeoApiClient`. So my question is, what name should we choose for the `@QueryMap`? It is almost exclusively used to refine the results coming back from the API.

`@Body` varies between being of type `Any`, `List<Any>`, `Map<String, String>`, and `Map<String, Any>`, as well as the concrete types that fully encapsulate the values in a type safe way. The names differ between `parameters` and `body` as well as `postBody` and `patchBody` in the `VimeoApiClient`.

So... how can we consolidate these names and types? The differences are mostly leftover from the old `VimeoClient` and `VimeoService` which served as the basis for these classes.

@WhosNickDoglio @howardrigberg

Reply:
My personal suggestion would be to choose `body` as the single name used for `@Body` and change the type to `Any` for anything that isn't a fully enumerated body DTO. I'm not sure what the name for `@QueryMap` params should be.